packages/deployed-backend-client/src/deployed_backend_client_list_sandboxes.test.ts

```diff
@@ -0,0 +1,228 @@
+import { beforeEach, describe, it, mock } from 'node:test';
+import assert from 'node:assert';
+import {
+  CloudFormation,
+  DescribeStacksCommand,
+  ListStacksCommand,
+  ListStacksCommandInput,
+  StackStatus,
+} from '@aws-sdk/client-cloudformation';
+import { BackendDeploymentStatus } from './deployed_backend_client_factory.js';
+import { platformOutputKey } from '@aws-amplify/backend-output-schemas';
+import { DefaultBackendOutputClient } from './backend_output_client.js';
+import { DefaultDeployedBackendClient } from './deployed_backend_client.js';
+import { StackIdentifier } from './index.js';
+import { AmplifyClient } from '@aws-sdk/client-amplify';
+import { S3 } from '@aws-sdk/client-s3';
+import { DeployedResourcesEnumerator } from './deployed-backend-client/deployed_resources_enumerator.js';
+import { StackStatusMapper } from './deployed-backend-client/stack_status_mapper.js';
+import { ArnGenerator } from './deployed-backend-client/arn_generator.js';
+import { ArnParser } from './deployed-backend-client/arn_parser.js';
+
+const listStacksMock = {
+  NextToken: undefined,
+  StackSummaries: [
+    {
+      StackName: 'amplify-test-name-sandbox-testHash',
+      StackStatus: StackStatus.CREATE_COMPLETE,
+      CreationTime: new Date(0),
+      LastUpdatedTime: new Date(1),
+    },
+  ],
+};
+
+const getOutputMockResponse = {
+  [platformOutputKey]: {
+    payload: {
+      deploymentType: 'sandbox',
+    },
+  },
+};
+
+void describe('Deployed Backend Client list sandboxes', () => {
+  const mockCfnClient = new CloudFormation();
+  const mockS3Client = new S3();
+  const cfnClientSendMock = mock.method(mockCfnClient, 'send');
+  let deployedBackendClient: DefaultDeployedBackendClient;
+  const listStacksMockFn = mock.fn();
+  const mockBackendOutputClient = new DefaultBackendOutputClient(
+    mockCfnClient,
+    new AmplifyClient()
+  );
+  const getOutputMock = mock.method(mockBackendOutputClient, 'getOutput');
+  const returnedSandboxes = [
+    {
+      deploymentType: 'sandbox',
+      backendId: {
+        namespace: 'test',
+        name: 'name',
+        type: 'sandbox',
+        hash: 'testHash',
+      },
+      name: 'amplify-test-name-sandbox-testHash',
+      status: BackendDeploymentStatus.DEPLOYED,
+      lastUpdated: new Date(1),
+    },
+  ];
+
+  beforeEach(() => {
+    getOutputMock.mock.mockImplementation(
```

Comment:
These need not be in `beforeEach` I'm guessing, can be moved out to the `describe`